File: api-asn/csharp/src/api/asn/cms/ECertificateChoices.cs
using tr.gov.tubitak.uekae.esya.api.asn.x509;
using tr.gov.tubitak.uekae.esya.asn.attrcert;
using tr.gov.tubitak.uekae.esya.asn.cms;
using tr.gov.tubitak.uekae.esya.asn.x509;
namespace tr.gov.tubitak.uekae.esya.api.asn.cms
{
    public class ECertificateChoices : BaseASNWrapper<CertificateChoices>
    {
        public ECertificateChoices(CertificateChoices aObject)
            : base(aObject)
        {
        }

        public ECertificateChoices(byte[] aBytes)
            : base(aBytes, new CertificateChoices())
        {
        }

        public ECertificateChoices(ECertificate aCertificate)
            : base(new CertificateChoices())
        {

            setCertificate(aCertificate);
        }

        public int getType()
        {
            return mObject.ChoiceID;
        }

        public void setCertificate(ECertificate aValue)
        {
            mObject.SetElement(CertificateChoices._CERTIFICATE, aValue.getObject());
        }

        public void setExtendedCertificate(ExtendedCertificate aValue)
        {
            mObject.SetElement(CertificateChoices._EXTENDEDCERTIFICATE, aValue);
        }

        public void setAttributeCertificateV1(AttributeCertificateV1 aValue)
        {
            mObject.SetElement(CertificateChoices._V1ATTRCERT, aValue);
        }

        public void setAttributeCertificateV2(AttributeCertificate aValue)
        {
            mObject.SetElement(CertificateChoices._V2ATTRCERT, aValue);
        }

        public void setOtherCertificateFormat(OtherCertificateFormat aValue)
        {
            mObject.SetElement(CertificateChoices._OTHER, aValue);
        }

        public ECertificate getCertificate()
        {
            if (mObject.ChoiceID == CertificateChoices._CERTIFICATE)
                return new ECertificate((Certificate)mObject.GetElement());

            return null;
        }

        public ExtendedCertificate getExtendedCertificate()
        {
            if (mObject.ChoiceID == CertificateChoices._EXTENDEDCERTIFICATE)
                return (ExtendedCertificate)mObject.GetElement();

            return null;
        }

        public AttributeCertificateV1 getAttributeCertificateV1()
        {
            if (mObject.ChoiceID == CertificateChoices._V1ATTRCERT)
                return (AttributeCertificateV1)mObject.GetElement();

            return null;
        }

        public AttributeCertificate getAttributeCertificateV2()
        {
            if (mObject.ChoiceID == CertificateChoices._V2ATTRCERT)
                return (AttributeCertificate)mObject.GetElement();

            return null;
        }


        public OtherCertificateFormat getOtherCertificateFormat()
        {
            if (mObject.ChoiceID == CertificateChoices._OTHER)
                return (OtherCertificateFormat)mObject.GetElement();

            return null;
        }
    }
}

File: api-asn/src/tr/gov/tubitak/uekae/esya/api/asn/cms/ECertificateChoices.java
package tr.gov.tubitak.uekae.esya.api.asn.cms;

import tr.gov.tubitak.uekae.esya.api.asn.BaseASNWrapper;
import tr.gov.tubitak.uekae.esya.api.asn.x509.ECertificate;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.asn.attrcert.AttributeCertificate;
import tr.gov.tubitak.uekae.esya.asn.attrcert.AttributeCertificateV1;
import tr.gov.tubitak.uekae.esya.asn.cms.CertificateChoices;
import tr.gov.tubitak.uekae.esya.asn.cms.ExtendedCertificate;
import tr.gov.tubitak.uekae.esya.asn.cms.OtherCertificateFormat;
import tr.gov.tubitak.uekae.esya.asn.x509.Certificate;

public class ECertificateChoices extends BaseASNWrapper<CertificateChoices> 
{
	public ECertificateChoices(CertificateChoices aObject)
	{
		super(aObject);
	}
	
	public ECertificateChoices(byte[] aBytes) throws ESYAException
	{
		super(aBytes,new CertificateChoices());
	}
	
	public ECertificateChoices(ECertificate aCertificate)
	{
		super(new CertificateChoices());
		setCertificate(aCertificate);
	}
	
	public int getType()
	{
		return mObject.getChoiceID();
	}
	
	public void setCertificate(ECertificate aValue)
	{
		mObject.setElement(CertificateChoices._CERTIFICATE, aValue.getObject());
	}
	
	public void setExtendedCertificate(ExtendedCertificate aValue)
	{
		mObject.setElement(CertificateChoices._EXTENDEDCERTIFICATE, aValue);
	}
	
	public void setAttributeCertificateV1(AttributeCertificateV1 aValue)
	{
		mObject.setElement(CertificateChoices._V1ATTRCERT, aValue);
	}
	
	public void setAttributeCertificateV2(AttributeCertificate aValue)
	{
		mObject.setElement(CertificateChoices._V2ATTRCERT, aValue);
	}
	
	public void setOtherCertificateFormat(OtherCertificateFormat aValue)
	{
		mObject.setElement(CertificateChoices._OTHER, aValue);
	}
	
	public ECertificate getCertificate()
	{
		if(mObject.getChoiceID()==CertificateChoices._CERTIFICATE)
			return new ECertificate((Certificate) mObject.getElement());
		
		return null;
	}
	
	public ExtendedCertificate getExtendedCertificate()
	{
		if(mObject.getChoiceID()==CertificateChoices._EXTENDEDCERTIFICATE)
			return (ExtendedCertificate) mObject.getElement();
		
		return null;
	}
	
	public AttributeCertificateV1 getAttributeCertificateV1()
	{
		if(mObject.getChoiceID()==CertificateChoices._V1ATTRCERT)
			return (AttributeCertificateV1) mObject.getElement();
		
		return null;
	}
	
	public AttributeCertificate getAttributeCertificateV2()
	{
		if(mObject.getChoiceID()==CertificateChoices._V2ATTRCERT)
			return (AttributeCertificate) mObject.getElement();
		
		return null;
	}
	
	
	public OtherCertificateFormat getOtherCertificateFormat()
	{
		if(mObject.getChoiceID()==CertificateChoices._OTHER)
			return (OtherCertificateFormat) mObject.getElement();
		
		return null;
	}
	
}
